### PR TITLE
CPLAT-7471 Refactor Prop Type Tests

### PR DESCRIPTION
**Pull request by joebingham-wk**
## Motivation
  
The prop type tests for over_react were still using the outdated `propType` testing methods. Refactoring the tests keeps them up to date with the latest APIs and helps provide examples of how to use our testing utils.
## Changes
  
- Refactor any test the tests `propTypes`
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 
@kealjones-wk @greglittlefield-wf @sydneyjodon-wk @aaronlademann-wf 
### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart [hunk 1]
```diff
@@ -67,62 +44,70 @@ void main() {
...
+        }, logsPropRequiredError('ComponentTestProps.required'));
       });
-
     });
 
     group('throwing when a prop is required and set to null', () {
       test('on mount', () {
-        render((ComponentTest()
+        expect(() {
+          render((ComponentTest()
```
Comment (aaronlademann-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: Should this be changed to use `mount` like the other tests?

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart [hunk 2]
```diff
@@ -67,62 +44,70 @@ void main() {
...
     group('throwing when a prop is nullable and not set', () {
       test('on mount', () {
-        render((ComponentTest()
-          ..required = true
-          ..requiredAndLengthLimited = [1,2]
-        )());
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.nullable')]);
+        expect(() {
+          render((ComponentTest()
```
Comment (aaronlademann-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: Should this be changed to use `mount` like the other tests?

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart [hunk 3]
```diff
@@ -67,62 +44,70 @@ void main() {
...
-              ..nullable = true
-              ..requiredAndLengthLimited = [1,2]
-            )());
+        expect(() {
+          jacket.rerender((ComponentTest()
+            ..required = true
+            ..nullable = true
+            ..requiredAndLengthLimited = [1,2]
+          )());
+        }, logsNoPropTypeWarnings);
```
Comment (aaronlademann-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: Why do we need to do a re-render to assert that the test is setup correctly? There is a `mount` happening with the same props above, couldn't we assert the `logsNoPropTypeWarnings` there?

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart [hunk 4]
```diff
@@ -165,26 +156,25 @@ void main() {
...
     group('not throwing when a prop is nullable and set to null', () {
       test('on mount', () {
-       render((ComponentTest()
-          ..nullable = null
-          ..requiredAndLengthLimited = [1,2]
-          ..required = true
-        )());
-        expect(consoleErrors, isEmpty, reason: 'should not have outputted a warning but found: $consoleErrors');
+        expect(() {
+          render((ComponentTest()
```
Comment (aaronlademann-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: Should we use `mount` to match the other tests?

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart [hunk 5]
```diff
@@ -196,38 +186,42 @@ void main() {
...
-          ..nullable = null
-          ..requiredAndLengthLimited = [1,2]
-        )());
+        expect(() {
+          jacket.rerender((ComponentTest()
+            ..required = true
+            ..nullable = true
+            ..requiredAndLengthLimited = [1,2]
+          )());
+        }, logsNoPropTypeWarnings);
```
Comment (aaronlademann-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: Same question here - can the `logsNoPropTypeWarnings` assertion not be made using the return value of `mount` above?

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart [hunk 6]
```diff
@@ -196,38 +186,42 @@ void main() {
...
       });
     });
 
     group('when a consumer propType function is also provided', () {
       test('required fires', () {
-       render((ComponentTest()
+        expect(() {
+          render((ComponentTest()
```
Comment (aaronlademann-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: Should we use `mount`?

File: test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart [hunk 7]
```diff
@@ -11,86 +11,66 @@
...
-import 'dart:js';
 
 import 'package:over_react/over_react.dart';
+import 'package:over_react_test/over_react_test.dart';
 import 'package:test/test.dart';
-import 'package:react/react_client/react_interop.dart' show PropTypes;
 
 import '../../../../test_util/test_util.dart';
 
 part 'required_accessor_integration_test.over_react.g.dart';
```
Comment (aaronlademann-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart]: Maybe I missed it - what is the fundamental difference between these tests - and the ones within `constant_required_accessor_integration_test.dart`?

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart (hunk 3 above)
Comment (joebingham-wk) [reply test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: Yep, good catch - I did do that the hard way!

File: test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart (hunk 7 above)
Comment (joebingham-wk) [reply test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart]: Really good question. The main difference is in the annotation of the component being tested. One uses `@requiredProp` / `@nullableRequiredProp` and the other uses `@Accessor`. The tests are functionally the same though.

File: pubspec.yaml [hunk 8]
```diff
@@ -39,8 +39,9 @@ dev_dependencies:
   dart2_constant: ^1.0.0
   dart_dev: ^3.0.0
   dependency_validator: ^1.4.0
+  front_end: ^0.1.27
```
Comment (greglittlefield-wf) [inline pubspec.yaml]: Why was this added as a dependency?

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart [hunk 9]
```diff
@@ -23,137 +22,114 @@ part 'constant_required_accessor_integration_test.over_react.g.dart';
...
-
-        jacket.rerender((ComponentTest()
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
```
Comment (greglittlefield-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: #nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart [hunk 10]
```diff
@@ -23,137 +22,114 @@ part 'constant_required_accessor_integration_test.over_react.g.dart';
...
-        var jacket = mount((ComponentTest()
-            ..required = true
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
```
Comment (greglittlefield-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: #nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart [hunk 11]
```diff
@@ -23,137 +22,114 @@ part 'constant_required_accessor_integration_test.over_react.g.dart';
...
       test('on re-render', () {
-        var jacket = mount((ComponentTest()
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
```
Comment (greglittlefield-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: #nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```

File: test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart [hunk 12]
```diff
@@ -165,69 +141,67 @@ void main() {
...
```
Comment (greglittlefield-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart]: #nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```

File: test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart [hunk 13]
```diff
@@ -11,152 +11,124 @@
...
-        expect(consoleErrors, [contains('ComponentTestProps.required')]);
-        expect(consoleErrors, [contains('This Prop is Required for testing purposes.')]);
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
```
Comment (greglittlefield-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart]: #nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```

File: test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart [hunk 14]
```diff
@@ -11,152 +11,124 @@
...
-        var jacket = mount((ComponentTest()
-            ..required = true
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
```
Comment (greglittlefield-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart]: #nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```

File: test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart [hunk 15]
```diff
@@ -11,152 +11,124 @@
...
       test('on re-render', () {
-        var jacket = mount((ComponentTest()
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
```
Comment (greglittlefield-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart]: #nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```

File: test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart [hunk 16]
```diff
@@ -168,69 +140,67 @@ void main() {
...
```
Comment (greglittlefield-wf) [inline test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart]: #nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```

File: pubspec.yaml (hunk 8 above)
Comment (joebingham-wk) [reply pubspec.yaml]: It definitely has to do with build_vm_compilers `1.0.3`. That version (and greater) in particular creates a build error that complains about not having front_end. I'm not sure why the dependency tree is getting into a bad state though

Comment (greglittlefield-wf) [reply pubspec.yaml]: Gross. I'm fine if this is necessary since it's just a dev dependency, but would you add a comment explaining this so we know that it's safe to remove down the road once it starts working again?

Comment (joebingham-wk) [reply pubspec.yaml]: Yes sir, that's a good idea
